evalml/utils/gen_utils.py

```diff
@@ -219,8 +219,16 @@ def _rename_column_names_to_numeric(X):
     Returns:
         Transformed X where column names are renamed to numerical values
     """
-    name_to_col_num = dict((col, col_num) for col_num, col in enumerate(X.columns.values))
-    return X.rename(columns=name_to_col_num, inplace=False)
+    X_t = X
+    if isinstance(X, np.ndarray):
+        return X
+    if isinstance(X, ww.DataTable):
```

Comment:
Ah, the `rename` method isn't in the latest release yet, so this will have to do for now :( 

But yes, I think you're right--updating to preserve types, since that's important!